### PR TITLE
python3Packages.pycapnp fixes

### PR DESCRIPTION
**Pull request by lukeadams**
###### Motivation for this change
pycapnp was marked as broken. That is now resolved.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pkgconfig/default.nix [hunk 1]
```diff
@@ -19,11 +19,9 @@ buildPythonPackage rec {
 
   checkInputs = [ nose ];
 
-  nativeBuildInputs = [ pkgconfig ];
+  propagatedNativeBuildInputs = [ pkgconfig ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pkgconfig/default.nix:22]: Is this the non python pkg-config? Please rename this to have a dash.

File: pkgs/development/python-modules/pkgconfig/default.nix [hunk 2]
```diff
@@ -19,11 +19,9 @@ buildPythonPackage rec {
...
 
-  checkPhase = ''
-    nosetests
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pkgconfig/default.nix:25]: Why did you remove tests?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pkgconfig/default.nix (hunk 1 above)
Comment (lukeadams) [reply pkgs/development/python-modules/pkgconfig/default.nix:22]: Yup, and will do.

File: pkgs/development/python-modules/pkgconfig/default.nix (hunk 2 above)
Comment (lukeadams) [reply pkgs/development/python-modules/pkgconfig/default.nix:25]: Turns out the pypi distribution strips the test files. Also the project moved to pytest. I'll fix/rebase, then @ you when done.

Comment (lukeadams) [reply pkgs/development/python-modules/pkgconfig/default.nix:25]: I just removed the test overrides – there's an upstream issue to include the test script in the pypi dist so next version bump should see tests working hopefully. Building form source is an option, but it uses "prose" which I have no experience with.

**Comment by lukeadams** [comment]
@SuperSandro2000 Made changes as requested. I'm just going to leave the tests alone since building from source just for the tests is a lot of trouble.

File: pkgs/development/python-modules/pkgconfig/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/pkgconfig/default.nix:25]: I think we need to add ``doCheck = false;`` or otherwise it will run pytest with zero tests executed and to make sure it is importable please add ``pythonImportsCheck = [ "pkgconfig" ]``

File: pkgs/development/python-modules/pycapnp/default.nix [hunk 3]
```diff
@@ -1,34 +1,33 @@
...
 
-  buildInputs = [ capnproto cython ];
+  buildInputs = [ capnproto cython pkgconfig ];
 
-  # import setuptools as soon as possible, to minimize monkeypatching mayhem.
-  postConfigure = ''
-    sed -i '3iimport setuptools' setup.py
-  '';
+  # Tests disabled due to dependency on jinja and various other libraries.
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pycapnp/default.nix:26]:
```suggestion
  doCheck = false;
  pythonImportsCheck = [ "pycapnp" ];
```

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108088` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>python3Packages.pycapnp</li>
</ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108088` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>python3Packages.pycapnp</li>
</ul>
</details>
